site/app/templates/officeHoursQueue/QueueStatus.twig

```diff
@@ -190,8 +222,56 @@
   </div>
 {% endif %}
 <script>
-    function toggleSwitchQueueForm(){
+     {% if viewer.inQueue() %}
+        function queueMessage() {
+                if (Notification.permission === "granted") {
+                    // If it's okay let's create a notification
+                    let notification = new Notification("New Message From Office Hours Queue Staff");
+                }
+                // Otherwise, we need to ask the user for permission
+                else if (Notification.permission !== 'denied' || Notification.permission === "default") {
+                    Notification.requestPermission(function (permission) {
+                        if (permission === "granted") {
+                            let notification = new Notification("New Message From Office Hours Queue Staff");
+                        }
+                    });
+                //if notification permissions are denied
```

Comment:
I was able to get prompted but before I could do anything the page reloaded and the pop up asking for permission went away. Removing refresh could help here for example. Also this seems like it will keep asking permission every time (maybe it won't but it definitely looks like it will). I wonder if maybe we ask for permission of notifications as soon as they load the office hours queue page. If they denied it, maybe we put a button on the page asking them if they ever change their mind they can request again. Notifications could also be used for maybe lets say when you are picked for help or when you are removed from the queue (probably another PR but having the notification change would help there).

Reply:
I got rid of the page reload problem. This will not ask for your permission every time, if you allow notifications it will never ask you again and if you deny notifications it will never ask you again. I like the idea of adding a button to allow users to change notification preferences and sending notifications for those other 2 scenarios I will make a quick PR for that after this one.